### PR TITLE
Added non-randomized tests of matching FlowSchema rules

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR adds non-randomized tests of the logic for matching FlowSchemas against RequestDigests.  This complements the existing randomized tests.  This is about the API
Priority and Fairness feature, which is first appearing in the coming release.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
[KEP] https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
```

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
@jktomer

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/match_test.go [hunk 1]
```diff
@@ -76,3 +80,243 @@ func TestPolicyRules(t *testing.T) {
...
+			IsResourceRequest: true,
+			Path:              "/apis/gooda/v1/namespaces/goodn/goods",
+			Verb:              "goodv",
+			APIPrefix:         "apis",
+			APIGroup:          "gooda",
+			APIVersion:        "v1",
+			Namespace:         "goodn",
+			Resource:          "goods",
+			Name:              "eman",
+			Parts:             []string{"goods", "eman"}},
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/match_test.go]: nit: Can we maybe choose different words for the test strings?

**Pull request merged**